src/core/simulation_results.jl

```diff
@@ -639,6 +640,11 @@ function SimulationResults(path::AbstractString, execution = nothing; load_syste
         end
     end
 
+    status = deserialize_status(joinpath(execution_path, RESULTS_DIR))
+    if status != RunStatus.SUCCESSFUL
+        error("Results can only be read from a successful simulation: $status")
```

Comment:
we should just warn here or allow to override to recover some results before the fail